### PR TITLE
Refactor scrolling on worksheets ui

### LINKED ISSUE #2685: worksheet page doesn't scroll automatically anymore when use keyboard shortcuts
On a long worksheet, when you scroll up and down using arrow keys or k/j, the worksheet used to follow along, but now it doesn't anymore.

### PR DESCRIPTION
**Pull request by epicfaace**
Refactor scrolling on worksheets ui.
- Rather than string refs, I directly get elements through their IDs. This is better than using string refs, which are deprecated and can be buggy.
- Don't use keepPosInView -- use [scrollIntoView()](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) instead, which is simpler and faster.

Fixes #2685, related to #2712.

### CODE REVIEW

**Comment by yipenghe** [comment]
Seems to work nice when we release the key after pressing, but if we hold the key down, the behaviour is weird

File: frontend/src/components/worksheets/items/WorksheetItem.js [hunk 1]
```diff
@@ -140,7 +141,7 @@ class TableWorksheetRow extends React.Component {
         var item = this.props.item;
         var className = /*'type-worksheet' + */ this.props.focused ? ' focused' : '';
         return (
-            <tr className={className}>
+            <tr className={className} id={this.props.id}>
```
Comment (michaeldu1) [inline frontend/src/components/worksheets/items/WorksheetItem.js:144]: can pull out id with other props since you do it already

**Review by michaeldu1 — APPROVED** [review]

**Comment by michaeldu1** [comment]
I also cannot reproduce the behavior Yipeng mentioned. Tested on local branch and code lgtm!

File: frontend/src/components/worksheets/items/WorksheetItem.js (hunk 1 above)
Comment (epicfaace) [reply frontend/src/components/worksheets/items/WorksheetItem.js:144]: Not in the `render()` function, though. will just leave it as is

**Pull request merged**